README.TXT
WIDE - WxWidgets IDE for Inform 6
=================================

WIDE is an Integrated Development Environment (IDE) for the Inform 6 Language.

Wide is a win32 Editor for Inform6: is an Integrated Development Environment (IDE), written entirely in C++/WxWidgets, for the creation of text adventures based on Graham Nelson's Inform6 Language. With Wide it's possible to edit, compile and run a text adventure in the z-code and glulx formats.

Main Features:

* Written in C++ (using the wxWidgets Library)
* Wide runs on Win32 / Linux systems
* Wide is localized in spanish, english, italian, german and french (simply customizing its wide.ini file...)
* Auto-completion window, in edit mode
* Wide is just an exe file: no DLL, no virtual machines. Wide is a stand alone application.
* Wrap lines, show number of lines, code folding
* Inform/Glulx support: wide supports zcode (5 and 8 version), ulx, blb, bres
* Objects tree and Syntax highlight is customizable
* Wide is portable: it can be started from an USB key/pen

Authors:
* Alessandro Schillaci
* Paolo Lucchesi

License:
Wide is an open source project (GNU GPL License).

Thanks to:
Urbatain, Christof Menear and Eric Forgeot for the wide translation in spanish, german and french.
Josè Manuel, for hints, suggestions and testing


INSTALL

Install on Linux:

1) install runtime wxgtk2.8
2) make
3) in case of errors uncomment line starting by "WX_LIBS" (see the makefile)
you shouldn't use "--as-needed" flag, use "--no-as-needed" instead


Install on Windows:

1) install and compile wxwidgets (using static=0 monolithic=1)
2) install MinGW
3) compile stc oomponent
4) cd wide
5) edit makefile.gcc to fit to your paths
6) mingw32-make -f makefile.gcc




changelog.txt
CHANGELOG WIDE
==============

NF  = New Feature
FIX = Bug Fix

2016-09-10
- [NF] project menues (create, update, change "wide project files" inside the wide editor)
- [FIX] Fixed setSelection problem (when selecting from left tree's objects)
- [NF] Added Syntax Highlighting for *.res files

2016-08-02
- [FIX] merges from Paolo Lucchesi stc->ConvertEOLs(2)
- [FIX] merges from Paolo Lucchesi font colors

2016-08-01
- [NF] updated makefile to wxwidgets 3.1.0 version
- [FIX] error in function "createblb" with spaces in path

2014-09-03 
- makefile fix
- hotkeys management
- colour fix
- gblorb file creation

2014-08-03
- WIDE VERSION 1.2.2
- [FIX] fix blc error under linux
- [FIX] fix colour schema
- [NF]  added tooltips over buttons
- [NF]  provided a configurazion to use CodeLite to compile wide
- [NF] wide.ini file saved locally and in the user dir

2014-07-22
- WIDE VERSION 1.2.1
- [NF]  console scroll to end
- [FIX] error on singledit
 
2014-07-20
- WIDE VERSION 1.2.0
- updated wide sources to compile with wxwidgets 3.0.1

2014-07-16
- new git repository

2013-07-15
- WIDE VERSION 1.01.01
- fixed static dll

2013-07-15
- WIDE VERSION 1.01.00
- added language menu
- added InformATE Spanish syntax
- added checkbox for syntax for:
    OTHERKEYWORDS
    DIRECTIVES
    STATEMENTS
    INFORMATE
    CUSTOM1
    CUSTOM2
    CUSTOM3

2013-07-07
- WIDE 1.00.00

2013-07-06
- fixed new autompletion
- hardcoded hotkeys (removed hotkeys config from wide.ini file)
- change background for selected line
- fixed blc under linux (should use absolute path for RES file)
- Makelower to poplist words

2013-07-05
- fixed Makefile (to compile under Arch)
- fixed path with " under linux systems (now it compiles)
- fixed autocomplete window performance (under linux)

2008-08-29
- fixed compatibility with linux



prefs.h
//////////////////////////////////////////////////////////////////////////////
// File:        prefs.h
// Purpose:     STC test Preferences initialization
// Maintainer:  Wyo
// Created:     2003-09-01
// RCS-ID:      $Id: prefs.h,v 1.1 2008-03-20 10:37:56 schillacia Exp $
// Copyright:   (c) wxGuide
// Licence:     wxWindows licence
//////////////////////////////////////////////////////////////////////////////

#ifndef _PREFS_H_
#define _PREFS_H_

//----------------------------------------------------------------------------
// informations
//----------------------------------------------------------------------------


//----------------------------------------------------------------------------
// headers
//----------------------------------------------------------------------------

//! wxWidgets headers

//! wxWidgets/contrib headers
#include "wx/stc/stc.h"  // styled text control

//! application headers


//============================================================================
// declarations
//============================================================================

//! general style types
#define mySTC_TYPE_DEFAULT 0

#define mySTC_TYPE_WORD1 1
#define mySTC_TYPE_WORD2 2
#define mySTC_TYPE_WORD3 3
#define mySTC_TYPE_WORD4 4
#define mySTC_TYPE_WORD5 5
#define mySTC_TYPE_WORD6 6

#define mySTC_TYPE_COMMENT 7
#define mySTC_TYPE_COMMENT_DOC 8
#define mySTC_TYPE_COMMENT_LINE 9
#define mySTC_TYPE_COMMENT_SPECIAL 10

#define mySTC_TYPE_CHARACTER 11
#define mySTC_TYPE_CHARACTER_EOL 12
#define mySTC_TYPE_STRING 13
#define mySTC_TYPE_STRING_EOL 14

#define mySTC_TYPE_DELIMITER 15

#define mySTC_TYPE_PUNCTUATION 16

#define mySTC_TYPE_OPERATOR 17

#define mySTC_TYPE_BRACE 18

#define mySTC_TYPE_COMMAND 19
#define mySTC_TYPE_IDENTIFIER 20
#define mySTC_TYPE_LABEL 21
#define mySTC_TYPE_NUMBER 22
#define mySTC_TYPE_PARAMETER 23
#define mySTC_TYPE_REGEX 24
#define mySTC_TYPE_UUID 25
#define mySTC_TYPE_VALUE 26

#define mySTC_TYPE_PREPROCESSOR 27
#define mySTC_TYPE_SCRIPT 28

#define mySTC_TYPE_ERROR 29

//----------------------------------------------------------------------------
//! style bits types
#define mySTC_STYLE_BOLD 1
#define mySTC_STYLE_ITALIC 2
#define mySTC_STYLE_UNDERL 4
#define mySTC_STYLE_HIDDEN 8

//----------------------------------------------------------------------------
//! general folding types
#define mySTC_FOLD_COMMENT 1
#define mySTC_FOLD_COMPACT 2
#define mySTC_FOLD_PREPROC 4

#define mySTC_FOLD_HTML 16
#define mySTC_FOLD_HTMLPREP 32

#define mySTC_FOLD_COMMENTPY 64
#define mySTC_FOLD_QUOTESPY 128

//----------------------------------------------------------------------------
//! flags
#define mySTC_FLAG_WRAPMODE 16

//----------------------------------------------------------------------------
// CommonInfo

struct CommonInfo {
    // editor functionality prefs
    bool syntaxEnable;
    bool foldEnable;
    bool indentEnable;
    // display defaults prefs
    bool readOnlyInitial;
    bool overTypeInitial;
    bool wrapModeInitial;
    bool displayEOLEnable;
    bool indentGuideEnable;
    bool lineNumberEnable;
    bool longLineOnEnable;
    bool whiteSpaceEnable;
};
extern const CommonInfo g_CommonPrefs;

//----------------------------------------------------------------------------
// LanguageInfo

struct LanguageInfo {
    wxChar *name;
    wxChar *filepattern;
    int lexer;
    struct {
        int type;
        const wxChar *words;
    } styles [STYLE_TYPES_COUNT];
    int folds;
};

extern const LanguageInfo g_LanguagePrefs[];
extern const int g_LanguagePrefsSize;

//----------------------------------------------------------------------------
// StyleInfo
struct StyleInfo {
    wxChar *name;
    wxChar *foreground;
    wxChar *background;
    wxChar *fontname;
    int fontsize;
    int fontstyle;
    int lettercase;
};

extern const StyleInfo g_StylePrefs[];
extern const int g_StylePrefsSize;

#endif // _PREFS_H_
